vpm.go

```diff
@@ -11,10 +12,23 @@ type vpm struct {
 	cfg plugins.Config
 }
 
-func (v *vpm) addPlugins() (err error) {
+func (v *vpm) addPlugins(pluginNames ...string) (err error) {
 	for _, pluginKey := range v.cfg.Plugins {
-		if err = v.addPlugin(pluginKey); err != nil {
-			return
+		if len(pluginNames) == 0 {
+			if err = v.addPlugin(pluginKey); err != nil {
+				return
+			}
+		} else {
+			// Filter
+			for _, name := range pluginNames {
```

Comment:
Minor perf improvement;

Move this filter above the loop for looping over plugin keys to build a pre-filtered list to iterate over instead.
O(n+m) instead of O(n*m).

Reply:
Good call, that was a refactor issue.

Reply:
Wait, no that's not how it works @cephalization 

The filter is a "contains" search. The way to improve performance is to convert pluginNames to a Map and then check for existence in O[1]

Reply:
Logic:

```
Iterate over plugins
  if filters present, check if should add, add if contained in filters
  if no filters present, add regardless
```

Reply:
v.cfg.Plugins is the full list
pluginNames are filters passed to vpm to limit build targets

```
vpm update contacts integrations voicemaildrop
```

Reply:
yeah sorry, I was writing that while dealing with my dog. I don't really think optimization is necessary here, I just saw for loop inside of for loop and decided to point it out.

Feel free to ignore!